mlflow/pytorch/__init__.py

```diff
@@ -886,6 +886,13 @@ def autolog(
     silent=False,
     registered_model_name=None,
     extra_tags=None,
+    model_checkpoint=True,
+    model_checkpoint_monitor="val_loss",
+    model_checkpoint_mode="min",
+    model_checkpoint_save_best_only=True,
+    model_checkpoint_save_weights_only=True,
+    model_checkpoint_every_n_epochs=None,
+    model_checkpoint_train_time_interval_S=600,
```

Comment:
this variable name is a bit strange. can we combine `model_checkpoint_every_n_epochs` and `model_checkpoint_train_time_interval_S` to`save_freq`? I find that's pretty clean and easy to use:
https://github.com/keras-team/keras/blob/fe2f54aa5bc42fb23a96449cf90434ab9bb6a2cd/keras/callbacks/model_checkpoint.py#L112

Reply:
One question:

The `save_freq` in keras checkpointing callback supports "do checkpoint after N batches (steps) execution", do we want to support this ? Or we only want to support epoch based checkpointing ?

My suggestion is we only support epoch based checkpointing, for batch based checkpointing, per-batch metric validation result is less accurate , and in pytorch-lightning, per-batch validation is not available except you log the metric with `on_step=True` i.e. `LightningModule.log(metric_name, value, on_epoch=True, on_step=True)`

Reply:
Demo notebook is attached in the PR description.

Reply:
`save_freq ` supports both per-epoch saving and per-N-steps saving. Both scenarios are commonly used based on my experience with the production team.

> in pytorch-lightning, per-batch validation is not available 

I think we can just use the callback hook? It should have the training stats available: https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.Callback.html#lightning.pytorch.callbacks.Callback.on_after_backward

Reply:
> I think we can just use the callback hook? It should have the training stats available

The issue is a bit different.

In ModelCheckpointCallback, if we use per-N-steps saving way, it needs to check "monitor" metric and **the metric must be updated on per step execution**, user need to add code in their module class like `log(metric_name, value, on_epoch=True, on_step=True)` to enable per-step updated metric.

But not a big problem, I can document it well.

Reply:
Yea if 2) is lightning's behavior, we can proceed with that one.